pyanaconda/users.py

```diff
@@ -168,6 +169,25 @@ def validatePassword(pw, user="root", settings=None):
 
     return (valid, strength, message)
 
+def check_name(in_name, ret_bool = False):
+    ign_dir = {}
+    for inc in os.listdir("/") + ["root", "home", "daemon"]:
+        ign_dir.update({inc: "Cannot set user name, reserved for system: "})
+    ign_dir.update({r"^[\d]": "User name cannot start with a digit: "})
+    ign_dir.update({r"[^A-Za-z0-9\s]": "User name cannot contain any non-ASCII character: "})
+
+    for inc in ign_dir:
+        tmp = regsearch(inc, in_name)
+        if tmp:
+            if ret_bool:
+                return False
+            else:
+                return ign_dir[inc] + "'%s'" % tmp.group()
+    if tmp == None:
+        return "Unknown issue with username"
+    if ret_bool:
+        return False
+
```

Comment:
In general I think the ``ret_bool`` parameter should be dropped and the function should just return the error message (or ``None`` if there's none) or, if the boolean value is really needed, it should return tuples like ``(True, None)`` or ``(False, error_msg)``.